extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/SendingInterceptor.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * Intercepts text or binary send from the {@link io.quarkus.websockets.next.runtime.WebSocketConnectionBase} connection.
+ */
+public sealed interface SendingInterceptor permits MetricsSendingInterceptor {
+
+    // following methods should mirror WebSocketConnectionBase sending methods
+
+    Runnable onSend(String text);
```

Comment:
I think that it would make a little more sense to change these methods to return `void` and let the call site create the `Runnable` where needed (ie. when the reactive pipeline is created).

Reply:
Right, I think you are right, fixed.